core/src/test/scala/integration/kafka/api/ProducerCompressionTest.scala

```diff
@@ -65,66 +63,58 @@ class ProducerCompressionTest(compression: String) extends ZooKeeperTestHarness
   @Test
   def testCompression() {
 
-    val props = new Properties()
-    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, TestUtils.getBrokerListStrFromServers(Seq(server)))
-    props.put(ProducerConfig.COMPRESSION_TYPE_CONFIG, compression)
-    props.put(ProducerConfig.BATCH_SIZE_CONFIG, "66000")
-    props.put(ProducerConfig.LINGER_MS_CONFIG, "200")
-    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer")
-    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer")
-    var producer = new KafkaProducer[Array[Byte],Array[Byte]](props)
-    val consumer = new SimpleConsumer("localhost", TestUtils.boundPort(server), 100, 1024*1024, "")
```

Comment:
The third parameter `100` is the socket timeout. 100ms is rather low, which could be the cause of the test failure. There are other tests with similar low timeout as well. Doesn't affect this PR though.